tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java

```diff
@@ -215,79 +280,43 @@ public void checkPlaceholders(CLDRFile cldrFileToTest) {
             String path = pathHeader.getOriginalPath();
             String value = cldrFileToTest.getStringValue(path);
             if (value == null) {
-                value = "?";
+                continue;
             }
             boolean containsMessagePattern = messagePlaceholder.reset(value)
                 .find();
             final Map<String, PlaceholderInfo> placeholderInfo = patternPlaceholders
                 .get(path);
             final PlaceholderStatus placeholderStatus = patternPlaceholders
                 .getStatus(path);
-            if (containsMessagePattern && placeholderStatus == PlaceholderStatus.DISALLOWED
```

Comment:
Revamp the old testing code, since it is not find-grained enough to be accurate.